app/controllers/finance/financial_links_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Finance::FinancialLinksController < Finance::BaseController
+
+  def show
+    @financial_link = FinancialLink.find(params[:id])
+
+    @items = []
+    @financial_link.financial_transactions.each do |ft|
+      @items << {
+        date: ft.created_on,
+        type: t('activerecord.models.financial_transaction'),
+        description: "#{ft.ordergroup.name}: #{ft.note}",
+        amount: ft.amount,
+        link_to: finance_ordergroup_transactions_path(ft.ordergroup)
+      }
+    end
+    @financial_link.invoices.each do |invoice|
+      @items << {
+        date: invoice.date || invoice.created_at,
+        type: t('activerecord.models.invoice'),
+        description: "#{invoice.supplier.name}: #{invoice.number}",
+        amount: invoice.amount,
+        link_to: finance_invoice_path(invoice)
+      }
+    end
+    @items.sort_by! { |item| item[:date] }
+  end
```

Comment:
I think this could be expressed clearer. One thing is use `map` and `+`. Maybe the functionality of gathering standard info from a `FinancialLink` relation belongs in `FinancialLink`?

Reply:
I use the functions for generating the links, which is not available at the model.